sunbird-project/controllers/project.js

```diff
@@ -116,30 +116,91 @@ const projectsList = async (req, res) => {
 	})
 }
 
-const createLocationReqBody = async (req, res, selectedConfig) => {
-	let targetedRoutePath = selectedConfig.targetRoute.path
-	const params = matchPathsAndExtractParams(selectedConfig.sourceRoute, req.originalUrl)
-	const targetRoute = pathParamSetter(targetedRoutePath, params)
-
-	let bodyData = {}
-	bodyData["request"] = {}
-	bodyData["request"]["filters"] = {}
-	if("_id" in req.body){
-		bodyData["request"]["filters"] = {
-			"id" : req.body._id
+const fetchLocationDetails = async (req, res, selectedConfig) => {
+	try{
+		// validate the body, if body is not present throw error
+		if(!(Object.keys(req["body"]).length > 0) || !(Object.keys(req["body"]["query"]).length>0)){
+			if(process.env.DEBUG_MODE == "true"){
+				console.log("req.body cannot be empty")
+			}
+			res.status(500).json("Internal Server Error")
 		}
-	}
-	if("code" in req.body){
-		bodyData["request"]["filters"] = {
-			"code" : req.body.code
+
+		// if passed api config has service value defined. We are getting the baseURl of that service from env of Interface service
+		if(selectedConfig.service){
+			req['baseUrl'] = process.env[`${selectedConfig.service.toUpperCase()}_SERVICE_BASE_URL`]
 		}
-	}
-	console.log(req.baseUrl, targetRoute, bodyData)
-	let response = await requesters.post(req.baseUrl, targetRoute, bodyData, {
-		"Authorization": `Bearer ${process.env.BEARER_TOKEN}`,
-	})
+		let targetedRoutePath = selectedConfig.targetRoute.path
+		const params = matchPathsAndExtractParams(selectedConfig.sourceRoute, req.originalUrl)
+		const targetRoute = pathParamSetter(targetedRoutePath, params)
+		
+		// prepare req.body to match sunbird location API req.body
+		let bodyData = {}
+		bodyData["request"] = {}
+		bodyData["request"]["filters"] = {}
+		if("_id" in req.body.query){
+			if(typeof req.body.query._id == "object"){
+				bodyData["request"]["filters"] = {
+					"id" : req.body.query._id["$in"]
+				}
+			}
+			else{
+				bodyData["request"]["filters"] = {
+					"id" : req.body.query._id
+				}
+			}
+		}
+		if("code" in req.body.query){
+			if(typeof req.body.query.code == "object"){
+				bodyData["request"]["filters"] = {
+					"code" : req.body.query.code["$in"]
+				}
+			}
+			else{
+				bodyData["request"]["filters"] = {
+					"code" : req.body.query.code
+				}
+			}
+		}
+		// fetch location details
+		let locationDetails = await requesters.post(req.baseUrl, targetRoute, bodyData, {
+			"Authorization": `Bearer ${process.env.BEARER_TOKEN}`,
```

Comment:
@Prajwal17Tunerlabs instead of calling it BEARER_TOKEN can we call it SUNBIRD_BEARER_TOKEN. Because this we have to add in interface service. in future if we have to add some other BEARER_TOKEN things will be clearer to understand